### PR TITLE
[MOBL-1399] Events module rewrite in Kotlin language

### PR DESCRIPTION
**Pull request by rahulrvp**
The event module is rewritten using Kotlin for an improved structure and performance. This code is covered with test cases.

### CODE REVIEW

**Review by ketanshikharebsft — APPROVED** [review]
LGTM

File: android-sdk/src/main/java/com/blueshift/Blueshift.java [hunk 1]
```diff
@@ -354,122 +359,94 @@ public void getLiveContentByCustomerId(@NonNull String slot, HashMap<String, Obj
...
+        try (BlueshiftNetworkRequestRepositoryImpl networkRequestRepository = new BlueshiftNetworkRequestRepositoryImpl(context)) {
+            try (BlueshiftEventRepositoryImpl eventRepository = new BlueshiftEventRepositoryImpl(context)) {
+                BlueshiftEventManager.INSTANCE.initialize(eventRepository, networkRequestRepository, BlueshiftLambdaQueue.INSTANCE);
+            } catch (Exception e) {
+                BlueshiftLogger.e(LOG_TAG, e);
             }
-        }
 
-        return result;
+            BlueshiftNetworkRepositoryImpl networkRepository = new BlueshiftNetworkRepositoryImpl();
```
Comment (rahulrvp) [inline android-sdk/src/main/java/com/blueshift/Blueshift.java:445]: If the line crashes, the catch block will be able to catch the exception and print the log.

File: android-sdk/src/main/java/com/blueshift/Blueshift.java [hunk 2]
```diff
@@ -354,122 +359,94 @@ public void getLiveContentByCustomerId(@NonNull String slot, HashMap<String, Obj
...
-                        HashMap<String, Object> extras = new HashMap<>();
-                        extras.put(BlueshiftConstants.KEY_PREVIOUS_APP_VERSION, storedAppVersionString);
-                        extras.put(BlueshiftConstants.KEY_APP_UPDATED_AT, CommonUtils.getCurrentUtcTimestamp());
+        // Cleanup any cached events by sending them to Blueshift.
+        BlueshiftExecutor.getInstance().runOnNetworkThread(() -> {
+            // Move any pending bulk events in the db to request queue.
+            BulkEventManager.enqueueBulkEvents(context);
+            // Sync the http request queue.
+            RequestQueue.getInstance().sync(context);
+        });
```
Comment (rahulrvp) [inline android-sdk/src/main/java/com/blueshift/Blueshift.java:432]: Keeping a flag to check if the queue was emptied before is a bit tricky and won't give accurate answer all the time. Both these calls have internal checks to make sure we are not running into expensive operations if the queue is empty.

File: android-sdk/src/test/java/com/blueshift/core/common/BlueshiftAPITest.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        assertEquals("${baseUrlForUS}api/v1/event", BlueshiftAPI.eventURL())
+        assertEquals("${baseUrlForUS}api/v1/bulkevents", BlueshiftAPI.bulkEventsURL())
+        assertEquals("${baseUrlForUS}track?key=value", BlueshiftAPI.trackURL("key=value"))
+    }
+
+    @Test
+    fun setDatacenter_whenSetToEU_shouldReturnCorrectURLForAllAPIsWithEUBaseUrl() {
+        BlueshiftAPI.setDatacenter(BlueshiftAPI.Datacenter.EU)
+
+        val baseUrlForUS = "https://api.eu.getblueshift.com/"
```
Comment (rahulrvp) [inline android-sdk/src/test/java/com/blueshift/core/common/BlueshiftAPITest.kt:29]: fix the naming

**Pull request merged**